docs/logging_snippets.py

```diff
@@ -0,0 +1,369 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Testable usage examples for Stackdriver Logging API wrapper
+
+Each example function takes a ``client`` argument (which must be an instance
+of :class:`google.cloud.logging.client.Client`) and uses it to perform a task
+with the API.
+
+To facilitate running the examples as system tests, each example is also passed
+a ``to_delete`` list;  the function adds to the list any objects created which
+need to be deleted during teardown.
+"""
+
+import time
+
+from google.cloud.logging.client import Client
+
+
+def snippet(func):
+    """Mark ``func`` as a snippet example function."""
+    func._snippet = True
+    return func
+
+
+def _millis():
+    return time.time() * 1000
+
+
+def do_something_with(item):  # pylint: disable=unused-argument
+    pass
+
+
+# pylint: disable=reimported,unused-variable,unused-argument
+@snippet
+def instantiate_client(_unused_client, _unused_to_delete):
+    """Instantiate client."""
+
+    # [START client_create_default]
+    from google.cloud import logging
+    client = logging.Client()
+    # [END client_create_default]
+
+    credentials = object()
+    # [START client_create_explicit]
+    from google.cloud import logging
+    client = logging.Client(project='my-project', credentials=credentials)
+    # [END client_create_explicit]
+# pylint: enable=reimported,unused-variable,unused-argument
+
+
+@snippet
+def client_list_entries(client, to_delete):  # pylint: disable=unused-argument
+    """List entries via client."""
+
+    # [START client_list_entries_default]
+    for entry in client.list_entries():  # API call(s)
+        do_something_with(entry)
+    # [END client_list_entries_default]
+
+    # [START client_list_entries_filter]
+    FILTER = "logName:log_name AND textPayload:simple"
```

Comment:
Single quotes? Also, why does `FILTER` need to be in caps? Instead use `filter_` (to avoid built-in collision)